File: ruby/lib/task_master/task_repository/csv_adapter.rb
module TaskMaster
  class TaskRepository
    class CsvAdapter
      def initialize(csv_file_path)
        @csv_file_path = csv_file_path
      end

      def list
        CSV.foreach(@csv_file_path, headers: true).map do |row|
          task_data(row)
        end
      end

      private

      def task_data(row)
        {
          id: row["Task ID"],
          name: row["Task Name"],
          assignee: row["Assignee"],
          due_on: row["Due Date"],
          status: row["Status"]
        }
      end
    end
  end
end

File: javascript/lib/taskRepository/CsvAdapter.js
const fs = require('fs');

class CsvAdapter {
	constructor(csvFilePath) {
		this.csvFilePath = csvFilePath;
	}

	list() {
		const data = fs.readFileSync(this.csvFilePath, 'utf8');
		const lines = data.split('\n').filter(Boolean);
		const results = lines.slice(1).map(line => this.#taskData(line.split(',')));
		return results;
	}

	#taskData(row) {
		return {
			id: row[0],
			name: row[1],
			assignee: row[2],
			dueOn: row[3],
			status: row[4]
		}
	}
}

module.exports = CsvAdapter;
